presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java

```diff
@@ -111,10 +115,24 @@ protected SchemaTableName getSchemaTableName(ResultSet resultSet)
     @Override
     protected String toSqlType(Type type)
     {
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            if (varcharType.getLength() <= MAX_IN_ROW_VARCHAR_LENGTH) {
+                return "varchar(" + varcharType.getLength() + ")";
+            }
+            else if (varcharType.getLength() <= 65535) {
+                return "text";
```

Comment:
I'd leave the support for `longtext`. Better not to put engine specific constraints into connectors when not necessary. We definitely need to support reading it (assuming the value is not actually too large).